Refactor/party/baek

changyeopbaek opened the pull request:
## PR 유형

- [ ] 새로운 기능 추가
- [ ] 버그 수정
- [x] CSS 등 사용자 UI 디자인 변경
- [x] 코드에 영향을 주지 않는 변경사항 (오타 수정, 탭 사이즈 변경, 변수명 변경)
- [ ] 코드 리팩토링
- [ ] 주석 추가 및 수정
- [ ] 문서 수정
- [ ] 테스트 추가, 테스트 리팩토링
- [ ] 빌드 부분 혹은 패키지 매니저 수정
- [ ] 파일 혹은 폴더명 수정/삭제

---

## PR 체크리스트

- [x] 커밋 메시지 컨벤션에 맞게 작성했습니다.
- [x] 변경 사항에 대한 테스트를 했습니다. (버그 수정/기능에 대한 테스트)

---

### PR 상세

---

## 이슈

resolves #

changyeopbaek committed:
[refactor] 문구 수정 및 불필요 코드 삭제

changyeopbaek committed:
Merge branch 'refactor/party/baek' of https://github.com/FRONTENDBOOTCAMP-14th/shuttlers.gg into refactor/party/baek

rhocci commented on src/app/party/find/page.tsx:
```diff
@@ -239,23 +239,37 @@ const PartyPage = () => {
-    };
+    if (participantError) {
+      setModalOpen(false);
+      await fetchParties();
```
이부분 코파일럿이 에러났을때 fetch하는건 낭비라고 그러는데.. 확인 부탁드립니다! 에러났을땐 fetch 전에 토스트 띄우고 early return하는게 낫지 않나 싶어요

changyeopbaek replied:
여기 파트는 확인해볼게요!

rhocci commented on src/app/party/find/page.tsx:
```diff
@@ -239,23 +239,37 @@ const PartyPage = () => {
       : [];
 
+    // 중복 참가자 방지
+    prevParticipants = prevParticipants.filter(
+      (p: any) => p.id !== currentUserId
+    );
```
여기도 참가자 insert 후에 중복을 검증하는 게 아니라, insert 전에 중복 검증해서 아예 추가 안되게 early return하는게 나을거같아요

changyeopbaek committed:
[refactor] 모임 참가 중복 방지 및 참가자 명단 업데이트 로직 개선

- 참가 시 프론트에서 중복 참가자 체크 후 insert 시도
- DB unique constraint 에러 발생 시 별도 안내 메시지 처리
- participants 컬럼 업데이트 시 중복 제거 후 추가
- 코드 가독성 및 에러 처리 분기 명확화
